File: tests/csharp/exceptions_advanced/custom_exception_class.cs
// vybe-test: csharp/exceptions_advanced/custom_exception_class
// origin: languages/csharp/tests/csharp/test_exceptions_advanced.rs

using static __Harness;

try {
    throw new AppException("not found", 404);
}
catch (AppException e) {
    __P((e.Message + " (" + e.Code + ")").ToString());
}
__Check("not found (404)");

class AppException : Exception {
    public int Code { get; set; }
    public AppException(string message, int code) : base(message) {
        Code = code;
    }
}

public static class __Harness {
    public static string __buf = "";
    public static void __P(string s) { __buf = __buf + s + "\n"; }
    public static void __Pr(string s) { __buf = __buf + s; }
    public static void __Check(string want) {
        if (__buf != want && __buf != want + "\n") {
            Console.WriteLine("FAIL: want [" + want + "] got [" + __buf + "]");
            throw new Exception("assertion failed");
        }
    }
}

File: tests/php/exception_chaining/custom_exception_class.php
<?php
// vybe-test: php/exception_chaining/custom_exception_class
// origin: languages/php/tests/php/test_exception_chaining.rs

function __vybe_check($got, $want) {
    // Match the Rust harness's normalisation: strip \r, then drop trailing
    // newlines (it split on "\n" and popped empty trailing elements).
    $got = str_replace("\r", "", $got);
    $got = rtrim($got, "\n");
    if ($got !== $want) {
        echo "FAIL: want [" . $want . "] got [" . $got . "]\n";
        throw new Exception("assertion failed");
    }
    // Replay the program's own output so running the file by hand still
    // behaves like the program it was extracted from.
    echo $got;
    if ($got !== "") {
        echo "\n";
    }
}

ob_start();

class DomainException2 extends RuntimeException {
    public function __construct(string $msg, private string $domain) {
        parent::__construct($msg);
    }
    public function getDomain(): string { return $this->domain; }
}
try { throw new DomainException2('err', 'payments'); }
catch (DomainException2 $e) { echo $e->getMessage() . ':' . $e->getDomain(); }

__vybe_check(ob_get_clean(), "err:payments");
